app/src/main/java/com/amaze/filemanager/filesystem/ssh/SshConnectionPool.java

```diff
@@ -169,27 +169,24 @@ public void expungeAllConnections() {
         });
     }
 
-    private SSHClient create(@NonNull String url) {
-        return create(Uri.parse(url));
-    }
-
     // Logic for creating SSH connection. Depends on password existence in given Uri password or
     // key-based authentication
-    private SSHClient create(@NonNull Uri uri) {
-        String host = uri.getHost();
-        int port = uri.getPort();
+    private SSHClient create(@NonNull String url) {
+        String host = url.substring(url.lastIndexOf('@')+1, url.lastIndexOf(':'));
```

Comment:
You are just assuming that the "URI" (a String) passed is in the correct format, without any checks. If a "malformed URI" (by this method's definition) gets to here, the error will be difficult to track down, maybe add a RegEx check (and crash with the String explicited in the exception message)? Or check manually if the String has the required parts?

Reply:
Surely a regex based implementation would be nice, but will need time to make a good regex - the regex I found with RFC2396 compliance doesn't parse URIs having `@`, `#` nicely, and that's the reason why I had to split the string by hand.

(That I think it's already a non-compliant URI)